src/vs/workbench/api/common/extHostLanguageFeatures.ts

```diff
@@ -579,6 +579,18 @@ class RangeFormattingAdapter {
 		}
 		return undefined;
 	}
+
+	async provideDocumentRangesFormattingEdits(resource: URI, ranges: IRange[], options: languages.FormattingOptions, token: CancellationToken): Promise<ISingleEditOperation[] | undefined> {
+		assertType(typeof this._provider.provideDocumentRangesFormattingEdits === 'function', 'INVALID invocation of `provideDocumentRangesFormattingEdits`');
+
+		const document = this._documents.getDocument(resource);
+		const _ranges = <Range[]>ranges.map(typeConvert.Range.to);
```

Comment:
`_` intentional?

Reply:
yeah, the argument is already called `ranges` and its being converted to API ranges